src/security/core/tests/common/test_utils.c

```diff
@@ -501,6 +501,9 @@ void rd_wr_init_fail(
       goto fail;
     dds_set_status_mask (*rd, DDS_SUBSCRIPTION_MATCHED_STATUS);
   }
+  dds_delete_qos (qos);
```

Comment:
These three lines don't add anything of value, it means the same code is executed under all circumstances. I prefer the simpler variant.